src/datasources/asset/AssetDataSource.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  DataFrameDTO,
+  DataQueryRequest,
+  DataSourceInstanceSettings,
+  TestDataSourceResponse,
+} from '@grafana/data';
+import { BackendSrv, getBackendSrv, getTemplateSrv, TemplateSrv } from '@grafana/runtime';
+import { DataSourceBase } from 'core/DataSourceBase';
+import {
+  AssetFilterProperties,
+  AssetModel, AssetQuery,
+  AssetQueryType,
+  AssetsResponse,
+} from './types';
+import { getWorkspaceName, replaceVariables } from "../../core/utils";
+import { SystemMetadata } from "../system/types";
+import { defaultOrderBy, defaultProjection } from "../system/constants";
+
+export class AssetDataSource extends DataSourceBase<AssetQuery> {
```

Comment:
This is more of a question .. but I am wondering , why aren't we using custom hooks for the datasources if we are using react components ? Are there some DI related reasons ?

Reply:
I follow the way the other plugins are written